Framework/Algorithms/inc/MantidAlgorithms/SampleCorrections/MCAbsorptionStrategy.h

```diff
@@ -38,8 +38,8 @@ class MANTID_ALGORITHMS_DLL MCAbsorptionStrategy {
                        size_t maxScatterPtAttempts);
   std::tuple<double, double> calculate(Kernel::PseudoRandomNumberGenerator &rng,
                                        const Kernel::V3D &finalPos,
-                                       double lambdaBefore,
-                                       double lambdaAfter) const;
+                                       double lambdaBefore, double lambdaAfter,
+                                       std::string &debugString) const;
```

Comment:
Could we instead move this into the class rather than forcing the caller to handle it?

I'm thinking that maybe we have:
```
class ... {
public:
  std::string getDebugString() const;

private:
  std:string debugString;  // Resets each time calculate is called
```

Reply:
I would need to remove the "const" qualifier from the MCAbsorptionStrategy::calculate function to do this (or I could leave the const qualifier in place and add the mutable qualifier to the debugString member). 
Passing the debugString as an output parameter seemed less bad than these two. What do you think?

Reply:
Since I had to remove the const qualifier from MCAbsorptionStrategy anyway for one of the other changes I have made this change now

Reply:
The latest build for this has failed on some of the Jenkins processes with a segmentation fault - I think it's because of this change relating to the debug string. The code that calls MCAbsorptionStrategy is multithreaded in MonteCarloAbsorption (see PARALLEL_FOR_IF). I think when there are several threads all trying to modify the same MCAbsorptionStrategy::debugString member it causes a problem. For this reason I wonder if it's best to revert to having it as a parameter on the Calculate function that is passed in\out - that way there is one instance per thread. What do you think?